### PR TITLE
Avoid changing global state in CI runs II

### PR DESCRIPTION
**Pull request by dougbu**
- move `dotnet` tools into `.dotnet/`
  - add `dotnet` tools to `$PATH`
- install `jq` in `.tools/`
- install `nginx` in `.tools/nginx`

### CODE REVIEW

**Review by JunTaoLuo — APPROVED** [review]

**Comment by dougbu** [comment]
@JunTaoLuo I changed this PR significantly since your review. I also added a commit avoiding some of the warnings that happen in every build. Please review again.

/cc @dotnet/aspnet-build

File: .azure/pipelines/jobs/default-build.yml [hunk 1]
```diff
@@ -232,29 +239,15 @@ jobs:
...
+          pathtoPublish: $(BuildDirectory)/${{ artifact.path }}
           ${{ if eq(artifact.name, '') }}:
             artifactName: artifacts-$(AgentOsName)-$(BuildConfiguration)
           ${{ if ne(artifact.name, '') }}:
             artifactName: ${{ artifact.name }}
           artifactType: Container
           parallel: true
 
-    - ${{ if eq(parameters.isTestingJob, true) }}:
-      - task: PublishTestResults@2
```
Comment (JunTaoLuo) [inline .azure/pipelines/jobs/default-build.yml:247]: Why is this removed?

File: .azure/pipelines/jobs/default-build.yml [hunk 2]
```diff
@@ -263,12 +256,3 @@ jobs:
           testResultsFiles: '**/artifacts/log/**/*.junit.xml'
           buildConfiguration: $(BuildConfiguration)
           buildPlatform: $(AgentOsName)
-      - task: PublishTestResults@2
```
Comment (JunTaoLuo) [inline .azure/pipelines/jobs/default-build.yml:266]: And this?

**Review by JunTaoLuo — APPROVED** [review]
Looks good, just a couple none blocking questions

File: .azure/pipelines/jobs/default-build.yml (hunk 2 above)
Comment (dougbu) [reply .azure/pipelines/jobs/default-build.yml:266]: Because TEST-junit-jupiter.xml also does not exist even in signalr-daily-tests runs. The `junit` result files I see under artifacts/logs/ have names like win32.signalr.junit.xml. In any case, the name seems more important in Helix test runs (see [last few lines of src/SignalR/clients/java/signalr/signalr.client.java.Tests.javaproj](https://github.com/dotnet/aspnetcore/blob/3132e5c6cee173bc025f36b2cfa3c53161370ca1/src/SignalR/clients/java/signalr/signalr.client.java.Tests.javaproj#L87-L88)). It's in Helix runs that I see e.g. `com.microsoft.signalr.HubConnectionTest` tests reported.

@BrennanConroy please confirm this file won't exist on our build agents, just on Helix machines.

Comment (BrennanConroy) [reply .azure/pipelines/jobs/default-build.yml:266]: The Java tests moved to Helix and should only be running there now. However, I think I still saw signalr-daily running them (which it shouldn't), so this is fine.
